nc/source/graphics/resources/GraphicsResources.cpp

```diff
@@ -122,14 +127,79 @@ namespace nc::graphics
         return m_accessors.at(uid);
     }
 
+    void TexturesData::AddTexture(Graphics* graphics, Texture texture)
+    {
+        auto index = m_accessors.size();
+        m_accessors.emplace(texture.uid, index);
+        m_textures.push_back(std::move(texture));
+
+        UpdateTextures(graphics);
+    }
+
+    void TexturesData::AddTextures(Graphics* graphics, std::vector<Texture> textures)
+    {
+        for (auto& texture : textures)
+        {
+            auto index = m_accessors.size();
+            m_accessors.emplace(texture.uid, index);
+            m_textures.push_back(std::move(texture));
+        }
+
+        UpdateTextures(graphics);
+    }
+
+    void TexturesData::UpdateTextures(Graphics* graphics)
+    {
+        m_sampler = graphics->GetBasePtr()->CreateTextureSampler();
+
+        std::array<vk::WriteDescriptorSet, 2> writes;
+        vk::DescriptorImageInfo samplerInfo = {};
+        samplerInfo.sampler = m_sampler.get();
+
+        if (!m_texturesInitialized)
+        {
+            m_imageInfos.reserve(m_maxTexturesCount);
```

Comment:
Why not use the count + value vector constructor here instead of reserve/push_back?

m_imageInfos = std::vector<whatever>(m_maxTexturesCount, m_textures.at(0).imageInfo);

Reply:
Also, why can't this be done in the constructor? Maybe I'm missing something with the m_textures[0].imageInfo bit...